viscy/representation/engine.py

```diff
@@ -137,9 +196,13 @@ def validation_step(self, batch: TripletSample, batch_idx: int) -> Tensor:
         anchor_projection = self(anchor)
         negative_projection = self(neg_img)
         positive_projection = self(pos_img)
-        loss = self.loss_function(
-            anchor_projection, positive_projection, negative_projection
-        )
+        if isinstance(self.loss_function, NTXentLoss):
+            # Note: we assume the two augmented views are the anchor and positive samples
+            loss = self.loss_function(anchor_projection, positive_projection)
```

Comment:
Since negative sample does not contribute to loss, the forward pass is not necessary (so is the data loading cost).